### ISSUE TITLE
Allow for IssueTracker specific attributes

### ISSUE THREAD

**Issue by ryanemerson** [jboss-set/aphrodite#46]
Could be achieved by utilising subclasses of Issue

**Comment by elguardian** [comment]
just to update the discussion regarding this enhacement. We agreed that instead of using inheritance we should implemente something more like an extension mechanism model.

1) create an interface ExtendedModel
2) add few calls to that interface List<String> findAttributes(), isAttributeSupported(String atrrName), <T>getAttribute(String attrName), getAttributeClass(String name) ?
3) make the domain model entity implements that interface.

**Comment by rpelisse** [comment]
I don't really like having the Issue class implements the ExtendedModel. Would you be OK with having a field in Issue instead:

`private ExtendedModel extendedModel;`

**Comment by elguardian** [comment]
well this is a  matter of concept... an entity is a extendedmodel; it does not have a extended model.

**Comment by rpelisse** [comment]
We can rename that, what I don't like is having all the methods of extendedModel inside the Issue POJO. It looked a bit messy and the class is already quite long. What about having a field called, I don't know, something like "trackerSpecificDatas" typed with the ExtendedModel interface (that we can also rename if you want).

**Comment by ryanemerson** [comment]
I don't think 4 additional methods is a big deal tbh. But if it really is an issue, you could create a Wrapper class that takes an issue as an argument, with an issue just exposing the map to the additional fields. For example:

``` java
AdditionalFields additionalFields = AdditionalFields.get(issue);
additionalFields.isAttributeSupported("example");
```

Personally I think we should just add the 4 extra methods to Issue.

**Comment by rpelisse** [comment]
hum..... OK, I don't like to be stubborn, but I'll go for the addtional fields approach if you don't mind guys, because I really don't like how the ExtendedModel looks. I'll try to push a PR with that today or Monday at latest.

**Comment by rpelisse** [comment]
@wolfc we mention this issue during our status call, what do you feel about the current extendedModel / AdditionalFields approach (as opposed to the subclass one originally proposed by Ryan and I) ?

**Comment by wolfc** [comment]
Do we have an example that requires the `AdditionalFields` approach?
I don't like to lose type-safety and simple compiler checks for nothing.

`AdditionalFiels.getAttribute("example")` is a definite no-go, at the least it should be `AdditionalFields.getAttribute("example", Example.class)`. And even then `issue.getExample()` trumps it.

**Comment by rpelisse** [comment]
This has been done and included since 0.5.3.
